src/providers/SettingContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+  PropsWithChildren,
+  ReactElement,
+  createContext,
+  useEffect,
+  useState,
+} from "react"
+import { darkTheme, lightTheme } from "../theme"
+import { CssBaseline } from "@mui/material"
+import { ThemeProvider } from "@mui/material"
+import { getCookie } from "../utils/helper"
+import useMediaQuery from "@mui/material/useMediaQuery"
+
+export type ThemeMode = "light" | "dark" | "system"
+export type SettingsContextProps = {
+  themeMode: ThemeMode
+  onChangeMode: (themeMode: ThemeMode) => void
+}
+
+const initialState: SettingsContextProps = {
+  themeMode: "light",
+  onChangeMode: () => undefined,
+}
+
+const SettingsContext = createContext(initialState)
+
+function SettingsProvider({
+  children,
+}: PropsWithChildren<unknown>): ReactElement {
+  const [mode, setMode] = useState<ThemeMode>("system")
+  const prefersDarkMode = (useMediaQuery("(prefers-color-scheme: dark)")
+    ? "dark"
+    : "light") as ThemeMode
+
+  useEffect(() => {
+    const initialMode = (getCookie("paletteMode") || "system") as ThemeMode
```

Comment:
We should leverage TS Enums to avoid "magic strings" throughout the whole app as it will give us a dev experience improvement in the long run.

https://eddieabbondanz.io/post/typescript/magic-strings/